src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -513,6 +501,84 @@ private void AddCentralTransitiveDependencyGroupsForPackageReference(PackageSpec
             }
         }
 
+        /// <summary>
+        /// Determines the <see cref="LibraryDependency" /> objects for the specified <see cref="RestoreTargetGraph" /> that represent the centrally defined transitive dependencies.
+        /// </summary>
+        /// <param name="targetGraph">The <see cref="RestoreTargetGraph" /> to get centrally defined transitive dependencies for.</param>
+        /// <param name="targetFrameworkInformation">The <see cref="TargetFrameworkInformation" /> for the target framework to get centrally defined transitive dependencies for.</param>
+        /// <param name="centralPackageTransitivePinningEnabled">A value indicating whether or not central transitive dependency version pinning is enabled.</param>
+        /// <returns>An <see cref="IEnumerable{LibraryDependency}" /> representing the centrally defined transitive dependencies for the specified <see cref="RestoreTargetGraph" />.</returns>
+        private IEnumerable<LibraryDependency> GetLibraryDependenciesForCentralTransitiveDependencies(RestoreTargetGraph targetGraph, TargetFrameworkInformation targetFrameworkInformation, bool centralPackageTransitivePinningEnabled)
+        {
+            foreach (GraphNode<RemoteResolveResult> node in targetGraph.Graphs.SelectMany(i => i.InnerNodes))
+            {
+                // Only consider nodes that are Accepted, IsCentralTransitive, and have a centrally defined package version
+                if (node?.Item == null || node.Disposition != Disposition.Accepted || !node.Item.IsCentralTransitive || !targetFrameworkInformation.CentralPackageVersions?.ContainsKey(node.Item.Key.Name) == true)
+                {
+                    continue;
+                }
+
+                CentralPackageVersion centralPackageVersion = targetFrameworkInformation.CentralPackageVersions[node.Item.Key.Name];
+                Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags = _includeFlagGraphs[targetGraph];
+
+                LibraryIncludeFlags suppressParent = LibraryIncludeFlags.None;
+
+                if (centralPackageTransitivePinningEnabled)
+                {
+                    // Centrally pinned dependencies are not directly declared but the PrivateAssets from the top-level dependency that pulled it in should apply to it also
+                    foreach (GraphNode<RemoteResolveResult> parentNode in FlattenParentNodes(node))
+                    {
+                        LibraryDependency parentDependency = targetFrameworkInformation.Dependencies.FirstOrDefault(i => i.Name.Equals(parentNode.Item.Key.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
Do we need to be combing the values from the intermediate nodes as well? 

For standard top level packages, we combine private assets and include/exclude assets to generate the includes in the nuspec. 

For these transitive packages we're only doing the private assets now. 


I think we should be considering the includes from the parents as well no? 

A (PrivateAssets compile) -> B (in the nuspec include compile,runtime only) -> C. 

C should end up with a runtime only correct?

Reply:
We decided to stop at the last explicit parent and just use that.  There's always one node that brings something in transitively and so we don't really need to combine anything.

Reply:
> We decided to stop at the last explicit parent and just use that. There's always one node that brings something in transitively and so we don't really need to combine anything.

I feel like this didn't align with my understanding. 

We still should be combining things in the multi root package scenario no?
If one suppresses it, but the other one doesn't and it gets lifted, then the package should show up.